### DISCUSSION TITLE
Create a PR template to help improve ease of reviewing

### DISCUSSION THREAD

**Discussion by Omnikar** [helix-editor/helix#10464] [General]
As @kirawi described in [this comment](https://github.com/helix-editor/helix/discussions/3815#discussioncomment-3672484), making a PR template to have contributors explain the functionality and motivation of their contribution could help make reviewing PRs easier/faster, which is something that seems to be becoming increasingly necessary with the rate at which PRs being opened.

**Comment by poliorcetics** [comment]
Proposed template

```markdown
## Why this change ?

<!---
Explain here why you want to add this change to Helix.

This can be as simple as  "Fixes #issue" or as complex as you need it to be, for refactoring or deeper issues with design tradeoffs.

For theme and UI changes, post screenshots of the new look (and the old one if relevant).

If your change does not match one of the predefined categories or need a longer explanation, remove the list and write your description.
--->

<!--- Add issue number in the format "Closes #issue" to make GitHub automatically close it --->
- [ ] Bug fix
- [ ] Theme changed/added
- [ ] Language changed/added (highlights, queries, injections, config in `languages.toml` ...)
- [ ] UI change
- [ ] Feature <!--- Will very often need a longer description and a doc update --->
- [ ] Refactor

-----

## Tasks (not applicable to all PRs)

If something is unneeded, just don't check it, if it was needed, either the CI or a reviewer will tell you 😄.

<!--- Especially needed if you introduce a new feature or language --->
- [ ] *(Doc)* I ran `cargo xtask docgen` to ensure the generated documentation is up-to-date
- [ ] *(Doc)* I updated the user-facing documentation under `book/`
- [ ] *(Themes)* I ran `cargo xtask themelint` <!--- For when you add/modify a new theme --->
- [ ] *(Languages)* I ran `cargo xtask query-check` <!--- For when you add/modify a new language --->

## Questions

<!--- Do you have points you're not sure about ? Add them here --->
```

<details><summary>Rendered</summary>

## Why this change ?




- [ ] Bug fix
- [ ] Theme changed/added
- [ ] Language changed/added (highlights, queries, injections, config in `languages.toml` ...)
- [ ] UI change
- [ ] Feature 
- [ ] Refactor

-----

## Tasks (not applicable to all PRs)

If something is unneeded, just don't check it, if it was needed, either the CI or a reviewer will tell you 😄.


- [ ] *(Doc)* I ran `cargo xtask docgen` to ensure the generated documentation is up-to-date
- [ ] *(Doc)* I updated the user-facing documentation under `book/`
- [ ] *(Themes)* I ran `cargo xtask themelint` 
- [ ] *(Languages)* I ran `cargo xtask query-check` 

## Questions


</details>

**Comment by helmesjo** [comment]
Should this perhaps get a bump? Looking through PRs (eg. when reading release logs) is quite hilariously unsatisfactory. Like it's not just low effort, it's literally zero effort in a lot of them (description is _blank_)... (:

One straight forward template I added for my team is:

```markdown
# TLDR
Updated theme `dark_plus` with 
correct highlight color.
## Problem
The color for highlights did 
not match the original theme.
**<image with old color>**
## Solution
Updated highlight color from 
**<old color>** to **<new color>**.
**<image with new color>**
```

I intentionally chose a quite trivial example, which at the same time would bring huge improvements to the (in this case) totally blank PRs with similar changes.

Usually it takes about 20-30 sec to fill it in, a couple of minutes for more complex changes. Point being: The "extra time/to much effort" argument does not apply (I'm confident in saying that since we've pushed through hundreds of PRs using this template).

**Now the main benefit**: 
All PRs can be parsed (eg. just grab `TLDR`) and a (draft) release log can be generated automatically. And when you click on a link in the release notes for a change like this you'd actually get something useful: An image showing the _actual_ change.

Anyways, just a suggestion to get the ball rolling.
